poc/mechanism/mechanism.py

```diff
@@ -0,0 +1,31 @@
+"""
+Interface for DP mechanisms.
+
+A DP mechanism is responsible for sampling noise and adding noise to data.
+"""
+
+class DpMechanism:
+    # The data type applicable to this `DpMechanism`. Here we are assuming
+    # the data type doesn't change before and after noise.
+    DataType = None
+    DebiasedDataType = None
+
+    def add_noise(self, data: DataType) -> DataType:
+        """Add noise to a piece of input data. """
+        raise NotImplementedError()
+
+    def sample_noise(self, num_reps: int, dimension: int) -> DataType:
+        """
+        Sample noise for `num_reps` number of times, with `dimension`.
+        """
+        raise NotImplementedError()
+
+    def debias(self,
+               agg: DataType,
```

Comment:
What do you think?

Here and elsewhere.

```suggestion
               data: DataType,
```